lldb/test/API/tools/lldb-dap/evaluate/TestDAP_evaluate.py

```diff
@@ -101,9 +101,9 @@ def run_test_evaluate_expressions(
         if context == "repl":
             # In the repl context expressions may be interpreted as lldb
             # commands since no variables have the same name as the command.
-            self.assertEvaluate("var", r"\(lldb\) var\n.*")
+            self.assertEvaluate("list", r"\(lldb\) list\n.*")
         else:
-            self.assertEvaluateFailure("var")  # local variable of a_function
+            self.assertEvaluateFailure("list")  # local variable of a_function
```

Comment:
Why does evaluating `list` in `a_function` fail?

Reply:
In the Debug Adapter Protocol `evaluate` request (https://microsoft.github.io/debug-adapter-protocol/specification#Requests_Evaluate) the `context` changes whether we're looking for variables in the current frame or lldb commands. If the context is `repl` then we'll try to evaluate expressions as lldb commands, otherwise if the context is `hover`/`watch`/`variables` then we only look for variables in the current scope.

Reply:
Its also worth noting that we're at breakpoint 1 here, which is in `main` (line 25), not `a_function` (breakpoint 3, line 12), so the variable `list` is out of scope.